controller/tests/array_action/xiv/array_mediator_xiv_tests.py

```diff
@@ -256,6 +266,45 @@ def test_delete_snapshot_succeeds(self):
         self.mediator.client.cmd.snapshot_delete = Mock()
         self.mediator.delete_snapshot("snap-wwn")
 
+    def test_get_object_by_id_return_correct_snapshot(self):
+        snap_name = "snap"
+        snap_vol_name = "snap_vol"
+        xcli_snap = self._get_single_snapshot_result_mock(snap_name, snap_vol_name)
+        self.mediator.client.cmd.vol_list.return_value = xcli_snap
+        res = self.mediator.get_object_by_id(snap_name, "snapshot")
+        self.assertTrue(res.name == snap_name)
+        self.assertTrue(res.volume_name == snap_vol_name)
+
+    def test_get_object_by_id_return_correct_volume(self):
+        volume_name = "volume_name"
+        vol = utils.get_mock_xiv_volume(10, volume_name, "wwn")
+        ret = Mock()
+        ret.as_single_element = vol
+        self.mediator.client.cmd.vol_list.return_value = ret
+        res = self.mediator.get_object_by_id(volume_name, "volume")
+        self.assertTrue(res.name == volume_name)
+
+
+    def test_get_object_by_id_same_name_vol_exists_error(self):
+        snap_name = "snap"
+        snap_vol_name = ""
+        xcli_snap = self._get_single_snapshot_result_mock(snap_name, snap_vol_name)
+        self.mediator.client.cmd.vol_list.return_value = xcli_snap
+        with self.assertRaises(array_errors.ExpectedSnapshotButFoundVolumeError):
+            self.mediator.get_object_by_id(snap_name, "snapshot")
+
+    def test_get_object_by_id_returns_illegal_object_name(self):
```

Comment:
raise*

Reply:
done

Reply:
I don't see the fix

Reply:
can you check again? I change everywhere its "return" error to raise error

Reply:
I checked again now. definitely still there. and it is also easy to see in the conversation, when the relevant line changes, there should be an "outdated" label on top:
![image](https://user-images.githubusercontent.com/57443811/96119669-0004f280-0ef6-11eb-9523-d10fc0eb4183.png)

Reply:
you right, strange.. done now